container-registry/container-analysis/src/main/java/com/example/containeranalysis/CreateOccurrence.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.containeranalysis;
+
+// [START containeranalysis_create_occurrence]
+import com.google.cloud.devtools.containeranalysis.v1beta1.GrafeasV1Beta1Client;
+import com.google.containeranalysis.v1beta1.NoteName;
+import com.google.containeranalysis.v1beta1.ProjectName;
+import io.grafeas.v1beta1.Occurrence;
+import io.grafeas.v1beta1.Resource;
+import io.grafeas.v1beta1.vulnerability.Details;
+import java.io.IOException;
+import java.lang.InterruptedException;
+
+public class CreateOccurrence {
+  // Creates and returns a new Occurrence associated with an existing Note
+  public static Occurrence createOccurrence(String resourceUrl, String noteId, 
+      String occProjectId, String noteProjectId) throws IOException, InterruptedException {
+    // String resourceUrl = "https://gcr.io/project/image@sha256:foo";
+    // String noteId = "my-note";
+    // String occProjectId = "my-project-id";
+    // String noteProjectId = "my-project-id";
+    final NoteName noteName = NoteName.of(noteProjectId, noteId);
+    final String occProjectName = ProjectName.format(occProjectId);
+
+    Occurrence.Builder occBuilder = Occurrence.newBuilder();
+    occBuilder.setNoteName(noteName.toString());
+    Details.Builder detailsBuilder = Details.newBuilder();
+    // Details about the vulnerability instance can be added here
+    occBuilder.setVulnerability(detailsBuilder);
+    // Attach the occurrence to the associated image uri
+    Resource.Builder resourceBuilder = Resource.newBuilder();
+    resourceBuilder.setUri(resourceUrl);
+    occBuilder.setResource(resourceBuilder);
+    Occurrence newOcc = occBuilder.build();
+
+    // Initialize client that will be used to send requests. After completing all of your requests, 
```

Comment:
Is this thread safe?  Should I have one per instance? Or 1 / concurrent session?

Reply:
do you think that information is necessary here? This comment is replicated in each sample, so I'm trying to limit the amount of copy/pasted documentation. Maybe we could replace this  with a [link to the docs?](https://googleapis.github.io/google-cloud-java/google-cloud-clients/apidocs/index.html?com/google/cloud/devtools/containeranalysis/v1beta1/package-summary.html)

Reply:
SGTM

Reply:
The docs probably should be expanded on a bit.

Reply:
hmm the url for the client is a bit long to include in a code comment. Do you have any advice for how best to include it?

https://googleapis.github.io/googlecloudjava/googlecloudclients/apidocs/com/google/cloud/devtools/containeranalysis/v1beta1/package-summary.html

Reply:
Firebase dynamic links?  https://firebase.google.com/products/dynamic-links